### PR TITLE
Added navigation-timing/nav2_test_redirect_chain_xserver_final_original_origin

### PR DESCRIPTION
**Pull request by nicjansma**
Adds a test that:

* starts from a page on `web-platform.test/...`
* iframe added for `www.web-platform.test/...` (cross origin)
* which redirects to `www.web-platform.test/...` (cross origin)
* which redirects to `web-platform.test/...` (same as original origin)

In this case, because of the cross-origin redirects, `redirectCount`, `redirectStart` and `redirectEnd` should all be `0`.

### CODE REVIEW

**Comment by nicjansma** [comment]
CC @yoavweiss @npm1

File: navigation-timing/nav2_test_redirect_chain_xserver_final_original_origin.html [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        <title>Navigation Timing 2 WPT</title>
+        <link rel="author" title="Google" href="http://www.google.com/" />
+        <link rel="help" href="http://www.w3.org/TR/navigation-timing-2/#sec-PerformanceNavigationTiming"/>
+        <script src="/resources/testharness.js"></script>
+        <script src="/resources/testharnessreport.js"></script>
+        <script src="/common/utils.js"></script>
+        <script>
+            setup({ single_test: true });
+
+            function verifyTimingEventOrder(eventOrder, timingEntry) {
```
Comment (npm1) [inline navigation-timing/nav2_test_redirect_chain_xserver_final_original_origin.html:14]: Unused function?

**Review by npm1 — COMMENTED** [review]
Looks great!

File: navigation-timing/nav2_test_redirect_chain_xserver_final_original_origin.html (hunk 1 above)
Comment (nicjansma) [reply navigation-timing/nav2_test_redirect_chain_xserver_final_original_origin.html]: Yep!  Copy/paste from `nav2_test_redirect_chain_xserver_partial_opt_in.html` so also removed from there

**Review by npm1 — APPROVED** [review]

**Comment by sefeng211** [comment]
Question: `redirect-opt-in.py` sets `TAO` header for the cross-origin redirects, shouldn't it allow timings to be returned?

**Comment by npm1** [comment]
Looking at the deprecated interface, it seems to imply that the redirects should not be set for cross origin redirects (regardless of TAO). But I don't know where the `create the navigation timing entry` is called, do you know @yoavweiss  or @noamr ?

**Comment by noamr** [comment]
> Looking at the deprecated interface, it seems to imply that the redirects should not be set for cross origin redirects (regardless of TAO). But I don't know where the `create the navigation timing entry` is called, do you know @yoavweiss or @noamr ?

It is called from the HTML spec.

For navigation timing, `redirectCount` is set to zero if there are any cross-origin no-cors redirects [here](https://html.spec.whatwg.org/#navigating-across-documents:navigation-params-has-cross-origin-redirects-4).

The resource timing properties (`redirectStart` and `redirectEnd`) are handled in FETCH with the same TAO rules as any resource.

**Comment by npm1** [comment]
Ok thanks, does that answer your question @sefeng211 ? I also filed https://github.com/w3c/navigation-timing/issues/157 because it is really hard to find answers in this spec now.

**Comment by sefeng211** [comment]
So er....this test was written based on the deprecated interface, but it should be interpreted based on the current Fetch spec right...Am I misunderstanding something? 

If so, should I read it as, it returns 0 because the `redirectCount` is 0? I am still not sure why we should return 0 for `redirectStart` in this case. 

Sorry! I am having a hard time to understand the spec..

**Comment by npm1** [comment]
Ok, so I think the spec is currently following what Resource Timing is doing, which is to set it depending on TAO checks. In particular this test and the spec seem at odds. I filed https://github.com/w3c/navigation-timing/issues/158 to find out the answer, apologies for the mess in this spec!

**Comment by noamr** [comment]
> Ok, so I think the spec is currently following what Resource Timing is doing, which is to set it depending on TAO checks. In particular this test and the spec seem at odds. I filed [w3c/navigation-timing#158](https://github.com/w3c/navigation-timing/issues/158) to find out the answer, apologies for the mess in this spec!

Right. something is actually missing in the spec. I'm on it, thanks for capturing it.

**Pull request merged**